### PR TITLE
Fix name of the bundle in metadata.yaml

### PR DESCRIPTION
**Pull request by taurus-forever**
## Proposal
Avoid confusing on `charmcraft list-lib` output (due to wrong name in metadata.yaml):
```
>> charmcraft list-lib
> Library name    API    Patch
> postgresql      0      7
> postgresql_tls  0      4
```

## Release Notes
Fix bundle name in metadata.yaml.

## Testing
Testing on GH Actions.

### CODE REVIEW

**Comment by dragomirp** [comment]
I think `lib_path` and references to it need to be removed in `tox.ini`

**Comment by taurus-forever** [comment]
> I think `lib_path` and references to it need to be removed in `tox.ini`

I have removed the second commit and keep name fix only.
There are tests dependencies on the library in tests/integration/helpers/helpers.py:

```
from charms.pgbouncer_k8s.v0 import pgb
...
async def get_cfg(ops_test: OpsTest, unit_name: str) -> pgb.PgbConfig:
    """Gets pgbouncer config from pgbouncer container."""
    cat = await cat_file_from_unit(ops_test, INI_PATH, unit_name)
    return pgb.PgbConfig(cat)
```

Up to the team refactor this or keep.

**Review by delgod — APPROVED** [review]

**Review by marceloneppel — APPROVED** [review]

File: tests/integration/helpers/helpers.py [hunk 1]
```diff
@@ -9,18 +9,17 @@
...
 
 
-def get_leader(ops_test, application_name) -> Unit:
-    """Gets the leader unit for the given app name."""
+async def get_leader(ops_test, application_name) -> str:
+    """Gets the leader unit name for the given app name."""
     for unit in ops_test.model.applications[application_name].units:
-        if unit.is_leader_from_status():
-            return unit
+        if await unit.is_leader_from_status():
```
Comment (marceloneppel) [inline tests/integration/helpers/helpers.py:21]: This method is asynchronous.

File: tests/integration/helpers/helpers.py [hunk 2]
```diff
@@ -223,7 +222,7 @@ async def scale_application(ops_test: OpsTest, application_name: str, scale: int
         await ops_test.model.wait_for_idle(
             apps=[application_name],
             status="active",
-            timeout=600,
+            timeout=1000,
```
Comment (marceloneppel) [inline tests/integration/helpers/helpers.py:225]: Multiple tests were failing due to the shorter timeout.

File: tests/integration/bundle/test_bundle.py [hunk 3]
```diff
@@ -89,7 +89,7 @@ async def test_kill_pg_primary(ops_test: OpsTest):
     connstr = finos_databag.get("master", None)
     assert connstr, f"databag incorrectly populated, \n {finos_databag}"
 
-    pgb_unit = get_leader(ops_test, PGB).name
+    pgb_unit = await get_leader(ops_test, PGB)
```
Comment (marceloneppel) [inline tests/integration/bundle/test_bundle.py:92]: This helper function became asynchronous due to the change in an of its internal calls.

File: tests/integration/bundle/test_tls_bundle.py [hunk 4]
```diff
@@ -21,8 +21,6 @@
 
 METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
 FINOS_WALTZ = "finos-waltz"
-PGB = METADATA["name"]
```
Comment (marceloneppel) [inline tests/integration/bundle/test_tls_bundle.py:24]: This was returning the name of the bundle (the field that was originally changed in this PR).

File: tests/integration/relations/test_backend_database.py [hunk 5]
```diff
@@ -48,7 +48,10 @@ async def test_deploy_bundle(ops_test: OpsTest):
         pgb_unit = ops_test.model.applications[PGB].units[0]
         logging.info(await get_app_relation_databag(ops_test, pgb_unit.name, backend_relation.id))
         wait_for_relation_removed_between(ops_test, PG, PGB)
-        await ops_test.model.wait_for_idle(apps=[PG, PGB], status="active", timeout=600),
+        await asyncio.gather(
+            ops_test.model.wait_for_idle(apps=[PG], status="active", timeout=600),
+            ops_test.model.wait_for_idle(apps=[PGB], status="blocked", timeout=600),
+        )
```
Comment (marceloneppel) [inline tests/integration/relations/test_backend_database.py:54]: After the relation is removed between PostgreSQL and PgBouncer charms, the PgBouncer charm enters into a blocked state, not active.

File: tests/integration/relations/pgbouncer_provider/test_pgbouncer_provider.py [hunk 6]
```diff
@@ -327,14 +327,12 @@ async def test_multiple_pgb_can_connect_to_one_backend(ops_test: OpsTest):
...
     )
-    async with ops_test.fast_forward():
-        await ops_test.model.wait_for_idle(apps=[pgb_secondary], status="active"),
 
     await ops_test.model.add_relation(f"{pgb_secondary}:{BACKEND_RELATION_NAME}", f"{PG}:database")
     wait_for_relation_joined_between(ops_test, PG, pgb_secondary)
 
     async with ops_test.fast_forward():
-        await ops_test.model.wait_for_idle(apps=APP_NAMES + [pgb_secondary])
+        await ops_test.model.wait_for_idle(apps=APP_NAMES + [pgb_secondary], status="active")
```
Comment (marceloneppel) [inline tests/integration/relations/pgbouncer_provider/test_pgbouncer_provider.py:335]: The new PgBouncer application become active only after the relation is added.

File: tests/integration/bundle/test_tls_bundle.py [hunk 7]
```diff
@@ -51,7 +49,7 @@ async def test_tls_encrypted_connection_to_postgres(ops_test: OpsTest):
     logs = await run_command_on_unit(
         ops_test, postgresql_primary_unit, "/charm/bin/pebble logs -n=all"
     )
+    username = f"{PGB}_user_{relation.id}_{ops_test.model.info.name}".replace("-", "_")
     assert (
-        f"connection authorized: user=relation_id_{relation.id} database=waltz"
-        " SSL enabled (protocol=TLSv1.2, cipher=ECDHE-RSA-AES256-GCM-SHA384, bits=256)" in logs
+        f"connection authorized: user={username} database=waltz SSL enabled" in logs
```
Comment (marceloneppel) [inline tests/integration/bundle/test_tls_bundle.py:54]: This was checking the wrong username. Also, I saw that the TLS version was updated to 1.3, so I changed the string to a more generic one in the assertion.

**Pull request merged**